### PR TITLE
Typed Protocols: new API

### PR DESCRIPTION
**Pull request by coot**
# Description

Using input-output-hk/typed-protocols#52 & input-output-hk/typed-protocols#61.

TODO:

* [x] update `cabal.project` once input-output-hk/typed-protocols#52 is published.
* [x] the same once input-output-hk/typed-protocols#61 is published.

## Performance

No performance regression was found when running the benchmarking cluster using
`cardano-node-9.1.0` as a base.  No regression was found when syncing mainnet.

# Checklist

### Quality
* [x] Commit sequence makes sense and have useful messages, see [ref][contrib#git-history].
* [x] New tests are added and existing tests are updated.
* [x] Self-reviewed the PR.

### Maintenance
* [x] Linked an [issue][link-issue] or added the PR to the current sprint of [`ouroboros-network`][project] project.
* [x] Added labels.
* [x] Updated changelog files.
* [ ] The documentation has been properly updated, see [ref][contrib#documentation].

[project]: https://github.com/orgs/IntersectMBO/projects/5/views/1
[link-issue]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=
[contrib#git-history]: https://github.com/IntersectMBO/ouroboros-network/blob/master/CONTRIBUTING.md#git-history
[contrib#documentation]: https://github.com/IntersectMBO/ouroboros-network/blob/master/CONTRIBUTING.md#documentation

### CODE REVIEW

**Comment by crocodile-dentist** [comment]
Not related to the PR directly but I guess we could sneak it here, there's a usage binding in the module chain-sync that is not referenced anywhere - maybe we can remove it completely?

**Comment by coot** [comment]
> Not related to the PR directly but I guess we could sneak it here, there's a usage binding in the module chain-sync that is not referenced anywhere - maybe we can remove it completely?

Which one?

**Comment by crocodile-dentist** [comment]
> > Not related to the PR directly but I guess we could sneak it here, there's a usage binding in the module chain-sync that is not referenced anywhere - maybe we can remove it completely?
> 
> Which one?

It's this one: https://github.com/IntersectMBO/ouroboros-network/blob/coot/tx-submission/ouroboros-network/demo/chain-sync.hs#L191-L194

File: ouroboros-network-protocols/CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@
   NumTxIdsToAck
 * Use `NumTxIdsToAck` newtype wrapper for number of unacked txid's
   in `txSubmissionInbound` and `txSubmissionOutbound`
+* Use `typed-protocols-0.2.0.0`.
```
Comment (crocodile-dentist) [inline ouroboros-network-protocols/CHANGELOG.md:34]:
```suggestion
* Use `typed-protocols-0.3.0.0`.
```

File: ouroboros-network-framework/src/Ouroboros/Network/Channel.hs [hunk 2]
```diff
@@ -143,8 +144,8 @@ mvarsAsChannel :: MonadSTM m
 mvarsAsChannel bufferRead bufferWrite =
     Channel{send, recv}
   where
-    send x = atomically (putTMVar bufferWrite x)
-    recv   = atomically (Just <$> takeTMVar bufferRead)
+    send x  = atomically (putTMVar bufferWrite x)
+    recv    = atomically (     Just <$>    takeTMVar bufferRead)
```
[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network-framework/src/Ouroboros/Network/Channel.hs:148]: This hunk is not necessary at all, I fixed by removing it.

[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network-framework/src/Ouroboros/Network/Channel.hs:148]: Fixed

**Review by crocodile-dentist — APPROVED** [review]
LGTM

**Pull request merged**